### PR TITLE
Update ruby version to 3.3.3

### PR DESCRIPTION
**Pull request by howard-e**
Updates the ruby version used to build to 3.3. Confirmed through my own forks and local testing that this should work within issues:

* [howard-e/wai-website#1](https://github.com/howard-e/wai-website/pull/1), check the latest [build log](https://github.com/howard-e/wai-website/actions/runs/9402837047/job/25897871546) to confirm everything builds as expected.

**Note**: Drafting because this is also dependent on https://github.com/w3c/wai-website-data/pull/159 being merged first so the Gemfile.lock can be updated.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ exclude:
   - "vendor"
 # Build settings
 markdown: kramdown
-kramwdown:
+kramdown:
```
Comment (howard-e) [inline _config.yml:34]: The [kramdown config rule](https://github.com/howard-e/wai-website/pull/1/files#diff-ecec67b0e1d7e17a83587c6d27b6baaaa133f42482b07bd3685c77f34b62d883) was [misspelled](https://jekyllrb.com/docs/configuration/markdown/#kramdown-processor) so I suppose wasn’t technically being applied.

Other changes to support this have already been applied https://github.com/w3c/wai-tutorials/pull/783

File: netlify.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ command = "git submodule update --init --remote && npm install && bundle exec je
 publish = "_site"
 
 [build.environment]
-RUBY_VERSION = "2.6.2"
```
Comment (remibetin) [inline netlify.toml:7]: @howard-e The Netlify build has failed. I guess it expects a known value from this file? https://github.com/rvm/rvm/blob/master/config/known_strings

Comment (howard-e) [reply netlify.toml:7]: Yep, pinned it to 3.3.3 instead of 3.3. It should build now

**Comment by howard-e** [comment]
@remibetin this has been updated with https://github.com/w3c/wai-website-data/pull/160 being merged.

**Pull request merged**